### DISCUSSION TITLE
Kubelogin how do I install this?

### DISCUSSION THREAD

**Discussion by clbarrett** [actions/virtual-environments#9386] [General questions]
Was wondering how exactly this works.

I've had to transition to self-hosted build agents as we are transitioning to private AKS cluster and really appreciate that this project is available.  One last hurdle I am facing is that the newest versions of the AKS cluster require kubelogin to be installed and configured to run kubectl commands.  What's the normal way of adding this?  Do I just need to provision a VM and then add it and then create a new image from it?

I am also curious how exactly the azure build agents work if kubelogin is not being installed on them.  My old clusters obviously don't need kubelogin to work.  And I can't exactly test it against my new private AKS clusters.  These images "are" what's being used by the azure build agents, right?

**Comment by bernhardriegler** [comment]
From these issues and discussions, currently it is suggested to install on runtime:

https://github.com/actions/runner-images/issues/7098
https://github.com/actions/runner-images/issues/7378
https://github.com/actions/runner-images/issues/1746
https://github.com/actions/runner-images/discussions/9723
https://github.com/actions/runner-images/discussions/5598
